### PR TITLE
 Laboratory work #1, Chistyakova Vasilisa - 19FPL2

### PR DESCRIPTION
**Pull request by vchistyakova**

### CODE REVIEW

**Comment by WhiteJaeger** [comment]
The deadline is not that far - consider pushing changes so that we can review them.

File: lab_1/main.py [hunk 1]
```diff
@@ -12,7 +11,20 @@ def tokenize(text: str) -> list:
     e.g. text = 'The weather is sunny, the man is happy.'
     --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
     """
-    pass
+
+    sentences = ''
+    text1 = text.lower()
+    slices = list(text1)
+    l = [".", ",", "!", "?", ";", ":", "*", "’", "$", "@", "*", "/", "-", "_", "+", "=", "(", ")", "1", "2", "3",
```
Comment (WhiteJaeger) [inline lab_1/main.py]: `l` is not very descriptive, right?

File: lab_1/main.py [hunk 2]
```diff
@@ -12,7 +11,20 @@ def tokenize(text: str) -> list:
...
+             "4", "5", "6", "7", "8", "9", "0"]
+    for y in slices:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: `y` is not that descriptive, right?

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_1/main.py [hunk 3]
```diff
@@ -12,7 +11,20 @@ def tokenize(text: str) -> list:
...
     """
-    pass
+
+    sentences = ''
+    text1 = text.lower()
+    slices = list(text1)
+    l = ["”", "“", "^", "|", "}", "{", "]", "[", "&", "%", "№", "#", "`", "`", "'", "<", ">", ".", ",", "!", "?", ";", ":", "*", "’", "$", "@", "*", "/", "-", "_", "+", "=", "(", ")", "1", "2", "3",
+             "4", "5", "6", "7", "8", "9", "0"]
+    for y in slices:
+        for y in l:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: Please, do not use same variable name inside loops,
You have two `y` variables - Python can be confused and act unexpectedly.

File: lab_1/main.py [hunk 4]
```diff
@@ -25,7 +37,32 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
     stop_words = ['the', 'is']
     --> ['weather', 'sunny', 'man', 'happy']
     """
-    pass
+    words = open('C:\\Users\\Acer\\2020-2-level-labs\\lab_1\\stop_words.txt', "r", encoding='utf-8')
```
Comment (WhiteJaeger) [inline lab_1/main.py]: You have an argument `stop_words` - you should work with it, i.e. filter the out from the `tokens` - another argument.

File: lab_1/main.py [hunk 5]
```diff
@@ -25,7 +37,32 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
+    stop_words = words.read().split()
+    sentence = ' '.join(stop_words)
+    slices = list(sentence)
+    l = ["'", ",", ".", "`", "`", "”", "“"]
+    bad_inputs_first = [{}, (), None, 9, 9.34, True, False]
+    bad_inputs_second = [{}, (), None, 9, 9.34, True, False]
+    empty = []
+    for x in slices:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: Please, do not use same variable name inside loops,
You have two `x` variables - Python can be confused and act unexpectedly.

Also, make their names more descriptive.

File: lab_1/main.py [hunk 6]
```diff
@@ -25,7 +37,32 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
+        for x in l:
+            if x in slices:
+                slices.remove(x)
+    line = ''.join(slices)
+    stop_words = line.split()
+    for z in tokens:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: Please, do not use same variable name inside loops,
You have two `z` variables - Python can be confused and act unexpectedly.

Also, make their names more descriptive.

File: lab_1/main.py [hunk 7]
```diff
@@ -25,7 +37,32 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
+        for z in stop_words:
+            if z in tokens:
+                tokens.remove(z)
+    for z in tokens:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: Please, do not use same variable name inside loops,
You have two `z` variables - Python can be confused and act unexpectedly.

Also, make their names more descriptive.

File: lab_1/main.py [hunk 8]
```diff
@@ -25,7 +37,32 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
+        if z in bad_inputs_first:
+            return empty
+        else:
+            return tokens
+    for a in stop_words:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: Please, do not use same variable name inside loops,
You have two `a` variables - Python can be confused and act unexpectedly.

Also, make their names more descriptive.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_1/main.py [hunk 9]
```diff
@@ -12,7 +11,18 @@ def tokenize(text: str) -> list:
     e.g. text = 'The weather is sunny, the man is happy.'
     --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
     """
-    pass
+
```
Comment (WhiteJaeger) [inline lab_1/main.py]: You need checks for input arguments, smth like:
`if not isinstance(input_arg_name, expected_arg_type):`
`do smth here`.

File: lab_1/main.py [hunk 10]
```diff
@@ -25,7 +35,24 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
     stop_words = ['the', 'is']
     --> ['weather', 'sunny', 'man', 'happy']
     """
-    pass
+
+    bad_inputs_first = [{}, (), None, 9, 9.34, True, False]
+    bad_inputs_second = [{}, (), None, 9, 9.34, True, False]
```
Comment (WhiteJaeger) [inline lab_1/main.py]: Please, do not drag test values here.

File: lab_1/main.py [hunk 11]
```diff
@@ -25,7 +35,24 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
+    for constituent in tokens:
+        for component in stop_words:
+            if component in tokens:
+                tokens.remove(component)
+    for constituent in tokens:
+        if constituent in bad_inputs_first:
+            return empty
+        else:
+            result = tokens
+            return result
```
Comment (WhiteJaeger) [inline lab_1/main.py]: I guess, it was for testing. Please, remove.

File: lab_1/main.py [hunk 12]
```diff
@@ -25,7 +35,24 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
+    for component in stop_words:
+        if component in bad_inputs_second:
+            res = tokens
+            return res
```
Comment (WhiteJaeger) [inline lab_1/main.py]: I guess, it was for testing. Please, remove.

File: lab_1/main.py [hunk 13]
```diff
@@ -36,7 +63,14 @@ def calculate_frequencies(tokens: list) -> dict:
     e.g. tokens = ['weather', 'sunny', 'man', 'happy']
     --> {'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
     """
-    pass
+    dictionary = {}
+    freq = 0
+    for b in tokens:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: `b` is not very descriptive, right?

File: lab_1/main.py [hunk 14]
```diff
@@ -36,7 +63,14 @@ def calculate_frequencies(tokens: list) -> dict:
...
+        freq = tokens.count(b)
+        dictionary[b] = freq
+        result = dictionary
```
Comment (WhiteJaeger) [inline lab_1/main.py]: You do this every time the `for`-loop is executed. I think, it should only be done once - after `for`-loop.

File: lab_1/main.py [hunk 15]
```diff
@@ -49,7 +83,28 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list:
...
+
+    frequent = list(diction.keys())
+    frequent.sort(reverse = True)
+    collection = []
+    for number in frequent:
+        collection.append(diction[number])
+
+    finished = []
+    for element in collection:
+        if len(element) == 1:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: What is this check for?

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_1/main.py (hunk 15 above)
Comment (vchistyakova) [reply lab_1/main.py]: It is used to make one list which consists of strings.

File: lab_1/main.py [hunk 16]
```diff
@@ -12,7 +11,21 @@ def tokenize(text: str) -> list:
     e.g. text = 'The weather is sunny, the man is happy.'
     --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
     """
-    pass
+    if not isinstance(text, str):
+        return False
```
Comment (WhiteJaeger) [inline lab_1/main.py]: You are expected to return a `list`.

File: lab_1/main.py [hunk 17]
```diff
@@ -12,7 +11,21 @@ def tokenize(text: str) -> list:
...
+
+    else:
```
Comment (WhiteJaeger) [inline lab_1/main.py]: No need for `else`-block here.

File: lab_1/main.py [hunk 18]
```diff
@@ -25,7 +38,13 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
     stop_words = ['the', 'is']
     --> ['weather', 'sunny', 'man', 'happy']
     """
-    pass
```
Comment (WhiteJaeger) [inline lab_1/main.py]: There should be checks for input arguments. Smth like:
`if not isinstance(input_arg_name, expected_type):`
`do something here`

File: lab_1/main.py [hunk 19]
```diff
@@ -36,7 +55,13 @@ def calculate_frequencies(tokens: list) -> dict:
     e.g. tokens = ['weather', 'sunny', 'man', 'happy']
     --> {'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
     """
-    pass
```
Comment (WhiteJaeger) [inline lab_1/main.py]: There should be checks for input arguments. Smth like:
`if not isinstance(input_arg_name, expected_type):`
`do something here`

File: lab_1/main.py [hunk 20]
```diff
@@ -49,7 +74,28 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list:
     top_n = 1
     --> ['happy']
     """
-    pass
```
Comment (WhiteJaeger) [inline lab_1/main.py]: There should be checks for input arguments. Smth like:
`if not isinstance(input_arg_name, expected_type):`
`do something here`

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]
You should add checks for input arguments, like in your first function and fix some comments that I have left.
Should you have any questions - contact me.

File: lab_1/main.py [hunk 21]
```diff
@@ -36,7 +53,14 @@ def calculate_frequencies(tokens: list) -> dict:
     e.g. tokens = ['weather', 'sunny', 'man', 'happy']
     --> {'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
     """
-    pass
+    if isinstance(tokens, list):
+        dictionary = {}
+        frequency = 0
```
Comment (WhiteJaeger) [inline lab_1/main.py]: There is no need to define this variable, you can just do it the the `for`-loop.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]
You should take a look at the automated checks here: https://travis-ci.com/github/fipl-hse/2020-2-level-labs/jobs/394809433
And see which tests are failing, fix them and push changes.
You can always run tests locally.
Should you have any questions - contact me.

**Comment by WhiteJaeger** [comment]
Open another PR with lab_2.

**Pull request closed without merging**